### ISSUE TITLE
on hinge loss

### ISSUE THREAD

**Issue by wangmr** [elikip/bist-parser#25]
[question 1] 
in mstlstm.py line 285-286:
    if rscores[goldLabelInd] < rscores[wrongLabelInd] + 1:
        lerrs.append(rexprs[wrongLabelInd] - rexprs[goldLabelInd])
I think hinge loss should be:
    if rscores[wrongLabelInd] < rscores[goldLabelInd] + 1:
         lerrs.append(1 + rexprs[goldLabelInd] - rexprs[wrongLabelInd])

[question 2] 
in mstlstm.py line 291:
    loss = [(exprs[h][i] - exprs[g][i]) for i, (h,g) in enumerate(zip(heads, gold)) if h != g] # * (1.0/float(e))
I think hinge loss should be:
   loss = [(1 + exprs[g][i] - exprs[h][i]) for i, (h,g) in enumerate(zip(heads, gold)) if h != g]

can any one help me to understand the code, thanks!
                                                                                                                                               mr wang
